src/main/java/pl/patrycja/ox/ui/Sequence.java

```diff
@@ -138,6 +142,41 @@ private void generateDiagonalDown(int size, int unbrokenLine) {
         writeToFile(stringBuilder, fileSequence);
     }
 
+    private void generateDraw(int size) {
```

Comment:
My suggestion would be that you will extract whole sequences generator in separate PR.
It should have names for each conditions and `free` numbers all around.